src/main/java/com/vadmack/mongodbtest/service/ProjectService.java

```diff
@@ -31,7 +32,7 @@
 
     public ProjectDto findById(Long id) {
         return repository.findById(id).map(project -> modelMapper.map(project, ProjectDto.class))
-                .orElse(new ProjectDto());
+                .orElseThrow(() -> new NotFoundException(String.format("Project with id=%d not found", id)));
```

Comment:
`orElseThrow(() -> new NotFoundException(String.format("Project with id=%d not found", id))` - this is a repeated code. I suggest to create a new method:
`Project getById(Long id)`
that would return an entity if it exists and thrown an exception if not.